addOns/authhelper/src/test/java/org/zaproxy/addon/authhelper/AuthHeaderTrackerUnitTest.java

```diff
@@ -0,0 +1,240 @@
+
+        // When
+        tracker.onHttpRequestSend(msg, HttpSender.PROXY_INITIATOR, null);
+
+        msg.getRequestHeader().setHeader(HttpRequestHeader.AUTHORIZATION, token2);
+        tracker.onHttpRequestSend(msg, HttpSender.AJAX_SPIDER_INITIATOR, null);
+
+        // Then
+        assertThat(tracker.getHostCount(), is(equalTo(1)));
+        assertThat(tracker.getTokenForHost("http://www.example.com"), is(equalTo(token1)));
+        assertAuthorization(msg, token2);
+    }
+
+    @Test
+    void shouldJustTrackLatestToken() throws Exception {
+        // Given
+        String token1 = "12345";
+        String token2 = "45678";
+        String token3 = "67890";
+        HttpMessage msg = new HttpMessage();
+        msg.setRequestHeader("GET http://www.example.com/test/ HTTP/1.1");
+
+        // When
+        msg.getRequestHeader().setHeader(HttpRequestHeader.AUTHORIZATION, token1);
+        tracker.onHttpRequestSend(msg, HttpSender.PROXY_INITIATOR, null);
+        msg.getRequestHeader().setHeader(HttpRequestHeader.AUTHORIZATION, token2);
+        tracker.onHttpRequestSend(msg, HttpSender.PROXY_INITIATOR, null);
+        msg.getRequestHeader().setHeader(HttpRequestHeader.AUTHORIZATION, token3);
+        tracker.onHttpRequestSend(msg, HttpSender.PROXY_INITIATOR, null);
+
+        // Then
+        assertThat(tracker.getHostCount(), is(equalTo(1)));
+        assertThat(tracker.getTokenForHost("http://www.example.com"), is(equalTo(token3)));
+    }
+
+    @Test
+    void shouldTrackTokensAcrossMultipleHosts() throws Exception {
+        // Given
+        String token1 = "12345";
+        HttpMessage msg1 = new HttpMessage();
+        msg1.setRequestHeader("GET http://www.example.com:9090/test/ HTTP/1.1");
+        msg1.getRequestHeader().setHeader(HttpRequestHeader.AUTHORIZATION, token1);
+        String token2 = "45678";
+        HttpMessage msg2 = new HttpMessage();
+        msg2.setRequestHeader("GET http://www.example.com/test/ HTTP/1.1");
+        msg2.getRequestHeader().setHeader(HttpRequestHeader.AUTHORIZATION, token2);
+        String token3 = "67890";
+        HttpMessage msg3 = new HttpMessage();
+        msg3.setRequestHeader("GET https://www.example.com/test/ HTTP/1.1");
+        msg3.getRequestHeader().setHeader(HttpRequestHeader.AUTHORIZATION, token3);
+
+        // When
+        tracker.onHttpRequestSend(msg1, HttpSender.PROXY_INITIATOR, null);
+        tracker.onHttpRequestSend(msg2, HttpSender.PROXY_INITIATOR, null);
+        tracker.onHttpRequestSend(msg3, HttpSender.PROXY_INITIATOR, null);
+
+        // Then
+        assertThat(tracker.getHostCount(), is(equalTo(3)));
+        assertThat(tracker.getTokenForHost("http://www.example.com:9090"), is(equalTo(token1)));
+        assertThat(tracker.getTokenForHost("http://www.example.com"), is(equalTo(token2)));
+        assertThat(tracker.getTokenForHost("https://www.example.com"), is(equalTo(token3)));
+    }
+
+    @Test
+    void shouldClearTokens() throws Exception {
```

Comment:
+WhenClearCalled

Reply:
Updated to add `onClear` but I'm not really sure it adds that much value :P